src/app/dataset-view/dataset.subscriptions.interface.ts

```diff
@@ -38,8 +43,30 @@ export interface DatasetHistoryUpdate {
     kind?: DatasetKind;
 }
 
+export type DatasetLineageBasics = DatasetBasicsFragment & { metadata: CurrentSourceFetchUrlFragment };
+
 export interface LineageUpdate {
-    nodes: DatasetBasicsFragment[];
-    edges: DatasetBasicsFragment[][];
-    origin: DatasetBasicsFragment;
+    nodes: DatasetLineageBasics[];
+    edges: DatasetLineageBasics[][];
+    origin: DatasetLineageBasics;
+}
+
+export interface LineageGraphNodeData {
+    nodeKind: LineageGraphNodeType;
+    nodeDataObject: LineageGraphDatasetNodeObject;
+}
+
+export enum LineageGraphNodeType {
+    Dataset = "dataset",
+    Source = "source",
+}
+
+export interface LineageGraphDatasetNodeObject {
+    id?: string;
+    name?: string;
+    kind?: DatasetKind;
+    isRoot?: boolean;
+    isCurrent?: boolean;
+    access?: string;
+    accountName?: string;
```

Comment:
Why all these properties need `?` Feels very wrong.

Reply:
Yes, fixed to required fields